src/shared/nuclearnet/tests/nuclearnet_proxy_parser.tests.ts

```diff
@@ -0,0 +1,170 @@
+import { NUClearNetPacket } from 'nuclearnet.js'
+
+import { Encoder } from '../nuclearnet_proxy_parser'
+import { Decoder } from '../nuclearnet_proxy_parser'
+import { TYPES } from '../nuclearnet_proxy_parser_socketio'
+import { Packet } from '../nuclearnet_proxy_parser_socketio'
+
+describe('NUClearNetProxyParser', () => {
+
+  it('Roundtrips regular socket.io packets', () => {
+
+    const e = new Encoder()
+    const d = new Decoder()
```

Comment:
Can these duplicate instantiations go in `beforeEach` blocks? e.g.

```ts
describe('NUClearNetProxyParser', () => {
  let e: Encoder

  beforeEach(() => {
    e = new Encoder()
  })

  it(''Roundtrips regular socket.io packets', () => {
    // something with e
  })
```

Reply:
Sure, still learning jest :P